.github/workflows/test-application.yaml

```diff
@@ -7,6 +7,11 @@ on:
             - '[0-9]+.x'
             - '[0-9]+.[0-9]+'
 
+# This stops all previous ci runs of a pull request, if a new commit has been pushed
+concurrency:
+    group: ${{ github.workflow }}-${{ github.head_ref }}
```

Comment:
i think `github.head_ref` is only available inside of pull requests. not sure what effect this has for workflow runs on the `2.1`/`2.2`/`2.x` branch. maybe its enough to use `github.ref`?
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions

Reply:
```suggestion
    group: ${{ github.workflow }}-${{ github.ref }}
```